litentry/core/assertion-build/src/a2.rs

```diff
@@ -23,79 +23,89 @@ extern crate sgx_tstd as std;
 #[cfg(all(not(feature = "std"), feature = "sgx"))]
 use crate::sgx_reexport_prelude::*;
 
-use crate::{CheckJoinDiscordResponse, Error, Result};
+use crate::{Error, IDHubberResponse, Result};
 
 use itc_rest_client::{
 	http_client::{DefaultSend, HttpClient},
 	rest_client::RestClient,
-	RestGet,
+	RestGet, RestPost,
 };
-use std::{format, str, string::String, time::Duration, vec};
+use std::{format, str, string::String, time::Duration};
 use url::Url;
 
 use litentry_primitives::ParameterString;
 
 const DISCORD_BASE_URL: &str = "https://47.57.13.126:8080/";
 const TIMEOUT: Duration = Duration::from_secs(3u64);
 
-pub fn build(guild_id: ParameterString, user_id: ParameterString) -> Result<()> {
+pub fn build(guild_id: ParameterString, handler: ParameterString) -> Result<()> {
 	let base_url = Url::parse(DISCORD_BASE_URL).unwrap();
 	let http_client = HttpClient::new(DefaultSend {}, true, Some(TIMEOUT), None, None);
 	let mut client = RestClient::new(http_client, base_url);
 
-	let path = format!(
-		"/discord/joined?guildid={:?}&userid={:?}",
-		guild_id.into_inner(),
-		user_id.into_inner()
+	let get_path = format!(
+		"/discord/joined?guildid={:?}&handler={:?}",
+		guild_id.clone().into_inner(),
+		handler.clone().into_inner()
 	);
-	let query = vec![];
 
-	let response: CheckJoinDiscordResponse = client
-		.get_with::<String, CheckJoinDiscordResponse>(path, query.as_slice())
+	let get_response: IDHubberResponse = client
+		.get::<String, IDHubberResponse>(get_path)
 		.map_err(|e| Error::Assertion2Error(format!("{:?}", e)))?;
 
 	log::debug!(
-		"get response: data: {:?}, message: {:?}, hasError: {:?}, msgCode: {:?}, success: {:?}",
-		response.data,
-		response.message,
-		response.has_errors,
-		response.msg_code,
-		response.success
+		"get_response: data: {:?}, message: {:?}, hasError: {:?}, msgCode: {:?}, success: {:?}",
+		get_response.data,
+		get_response.message,
+		get_response.has_errors,
+		get_response.msg_code,
+		get_response.success
 	);
 
 	// TODO:
 	// generate_vc(who, identity, ...)
 
+	// Assign ID-Hubber role:
+	let post_path = format!(
+		"/discord/assgin/idhubber?guildid={:?}&handler={:?}",
+		guild_id.into_inner(),
+		handler.into_inner()
+	);
```

Comment:
Should the assignment of `IDHubber` be triggered by our assertion-build code or by F/E after it verifies VC? I thought that was a subsequent action after VC is verified.

IMO our assertion-build should only verify or compute things, not trigger any actions -- this should be done by the VC consumer. From an architectural PoV these should be separated.

Let's clarify it with @jingleizhang and maybe IDHub colleagues. This is unrelated to this PR though.

I'm happy with the rest, PR approved, thanks.

Reply:
About the IDHubber role assignment, it's defined in #35 . 
There should two parts hold this role info: F/E and bot. It's only the bot here.
Let's have a short alignment tomorrow morning.

Reply:
Based on the definition of assertion2 (https://github.com/litentry/tee-worker/issues/35), there are two ways for us to let the others(e.g, Bot) know about the verification results:

1.  TEE triggers the action to others under/behind assertion-build flow
2. user presents a vc to others after the assertion-build flow

Let's have an alignment today morning about:
- whether these trigger mechanisms should be contained in TEE or not
- in assertion-build flow,  whether the codes here should trigger/notify external endpoints or not

Reply:
Let use solution 2:
 1. TEE only serves the computation; 
 2. VC will be sent to F/E; 
 3. F/E trigger IDHubber role assignment;
 4. F/E align with bot.